client/webserver/site/src/js/wallets.js

```diff
@@ -237,6 +240,20 @@ export default class WalletsPage extends BasePage {
     await this.newWalletForm.loadDefaults()
   }
 
+  async rescanWallet (assetID) {
+    if (app().haveAssetOrders(assetID)) {
```

Comment:
We usually just check this in **core**, where we iterate `(*dexConnection).hasActiveAssetOrders`. I guess by doing it this way, you allow the rescan to be forced via **rpcserver**?

Reply:
Correct.  I started with it in Core, but went to frontend because there are legit reasons to rescan even with swaps and I didn't want to close the door to doing that with rpcserver or even a "force rescan" dialog on the frontend.

Reply:
That reminds me, I've prepared a simplified version of the `dropwtxmgr` tool from the btcwallet repo: https://github.com/decred/dcrdex/commit/4a1fdb71cda620618085a874cdd1e5d648e3ff98

client/cmd/btcwdroptxs/main.go would be a fallback way to force a rescan after shutting down dexc.
It tests well and is built from our repo, so I might PR this.

Reply:
> Correct. I started with it in Core, but went to frontend because there are legit reasons to rescan even with swaps (might even be the only way to finish a swap, even to refund or auto-redeem if btcwallet failed to recognize your own swap, who knows?) and I didn't want to close the door to doing that with rpcserver or even a "force rescan" dialog on the frontend.

Maybe the most flexible way is to add a `force bool` argument to `(*Core).RescanWallet` and to the rpcserver method.

Reply:
I rejected that idea in planning this because the caller still needs to provide the value for that bool when calling.  For frontend, that still means a prompt, so options there are (a) prompt based on known order info as done here, (b) just try rescan with force=false and prompt based on a _specific error_ returned, or (c) a combination of those.

I suppose an advantage to an arg on Core.RescanWallet would be that you could never accidentally rescan with an active  order unless you explicitly said so, which could happen if you had some automatic order placement routine going on concurrently.  It would also make the rpc a bit safer, so I'll make that change, but to avoid a round trip http api rescan request for the frontend, I'd still like to have the JS check.

Reply:
It can just be hard-coded to `force = false` for **webserver** and `force = true` for **rpcserver** for now, and the option added to the UI in the future without changes to the API.

Reply:
> hard-coded to `force = false` for **webserver**

This is what I'm doing now for webserver, but making it an arg defaulting to false for rpcserver